src/main/java/com/google/devtools/build/lib/rules/repository/LocalRepositoryFunction.java

```diff
@@ -122,7 +122,15 @@ protected static FileValue getWorkspaceFile(RootedPath directory, Environment en
             directory.getRoot(),
             directory
                 .getRootRelativePath()
-                .getRelative(BuildFileName.WORKSPACE.getFilenameFragment()));
+                .getRelative(LabelConstants.WORKSPACE_DOT_BAZEL_FILE_NAME));
+
+    if (!workspaceRootedFile.asPath().exists()) {
+      workspaceRootedFile = RootedPath.toRootedPath(
```

Comment:
Logic is not trivial, so please extract to helper function.

Reply:
Done.